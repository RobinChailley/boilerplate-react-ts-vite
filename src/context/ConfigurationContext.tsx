import { ConfigContextValueType } from '@context/configurationContextTypes';
import getConfigContextValue from '@context/getConfigurationContextValue';
import { createContext, useContext } from 'react';

export const ConfigContext = createContext<ConfigContextValueType>(getConfigContextValue());

export const useConfig = () => useContext(ConfigContext);
