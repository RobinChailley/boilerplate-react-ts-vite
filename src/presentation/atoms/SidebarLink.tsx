import { useConfig } from '@context/ConfigurationContext';
import React from 'react';
import { NavLink } from 'react-router-dom';

interface SidebarLinkProps {
    alt: string;
    icon: string;
    title: string;
    to: string;
}

const classes = {
    container: 'atom',
    link: 'text-base p-2 flex items-center dark:text-white rounded hover:bg-gray-700',
    img: 'h-5 w-5',
    text: 'ml-4',
};

const SidebarLink: React.FC<SidebarLinkProps> = ({
    alt,
    icon,
    title,
    to,
}) => {
    const { translatorAdapter } = useConfig();
    const t = translatorAdapter.translate();

    return (
        <li className={classes.container}>
            <NavLink
                to={to}
                className={classes.link}
            >
                <img className={classes.img} src={icon} alt={alt}/>
                <span className={classes.text}>{t(title)}</span>
            </NavLink>
        </li>
    );
};

export default SidebarLink;
