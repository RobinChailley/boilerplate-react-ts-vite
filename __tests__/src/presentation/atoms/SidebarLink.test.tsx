import { ConfigContext } from '@context/ConfigurationContext';
import SidebarLink from '@presentation/atoms/SidebarLink';
import renderer from 'react-test-renderer';

import getConfigContextValue from '../../../mocks/context/getConfigurationContextValue';

describe('SidebarLink', () => {
  it('should render correctly', async () => {
    let render = renderer.create(
      <ConfigContext.Provider value={getConfigContextValue()}>
        <SidebarLink alt={''} icon={''} title={''} to={''} />
      </ConfigContext.Provider>,
    );

    const snapshot = render!.toJSON();
    expect(snapshot).toMatchSnapshot();
  });
});
