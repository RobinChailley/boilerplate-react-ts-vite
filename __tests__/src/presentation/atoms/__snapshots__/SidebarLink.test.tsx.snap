// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidebarLink should render correctly 1`] = `
<li
  className="atom"
/>
`;
